docs/content/concepts/dagster-pipes/pyspark.mdx

```diff
@@ -0,0 +1,328 @@
+    )
+
+    message_reader = PipesS3MessageReader(
+        client=s3_client,
+        bucket=bucket,
+        # the following setting will configure the Spark job to collect logs from the driver
+        # and send them to Dagster via Pipes
+        include_stdio_in_messages=True,
+    )
+
+    with dg.open_pipes_session(
+        context=context,
+        message_reader=message_reader,
+        context_injector=context_injector,
+    ) as session:
+```
+
+Notice how `PipesS3MessageReader` has `include_stdio_in_messages=True`. This setting will configure the Pipes message writer in the Spark job to collect logs from the Spark driver and send them to Dagster via Pipes messages.
+
+2. We will be using CLI arguments to pass the bootstrap information from Dagster to the Spark job. In other Pipes workflows this is typically done via environment variables, but setting them for Spark jobs can be complicated (the exact way of doing this depends on the Spark deployment) or not possible at all. CLI arguments are a convenient alternative. We will fetch them from the `session.get_bootstrap_cli_arguments` method. We pass these arguments to `spark-submit` along with a few other settings.
+
+```python file=/guides/dagster/dagster_pipes/pyspark/dagster_code.py startafter=end_pipes_session_marker endbefore=start_definitions_marker
+# prepare Pipes bootstrap CLI arguments
+
+        dagster_pipes_args = " ".join(
+            [
+                f"{key} {value}"
+                for key, value in session.get_bootstrap_cli_arguments().items()
+            ]
+        )
+
+        cmd = " ".join(
+            [
+                "spark-submit",
+                # change --master and --deploy-mode according to Spark setup
+                "--master",
+                "local[*]",
+                "--conf",
+                "spark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem",
+                # custom S3 endpoint for MinIO
+                "--conf",
+                "spark.hadoop.fs.s3a.endpoint=http://minio:9000",
+                "--conf",
+                "spark.hadoop.fs.s3a.path.style.access=true",
+                f"s3a://{bucket}/{s3_script_path}",
+                dagster_pipes_args,
+            ]
+        )
+
+        subprocess.run(
+            # we do not forward stdio on purpose to demonstrate how Pipes collect logs from the driver
+            cmd,
+            shell=True,
+            check=True,  # stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+
+        return session.get_results()
+```
+
+3. At last, let's add the new asset to `Definitions`:
+
+```python file=/guides/dagster/dagster_pipes/pyspark/dagster_code.py startafter=start_definitions_marker endbefore=end_definitions_marker
+defs = dg.Definitions(
+    assets=[pipes_spark_asset],
+)
+```
+
+## Step 2: Use Pipes in the PySpark job
+
+Call `open_dagster_pipes` in the PySpark job script to create a context that can be used to send messages to Dagster:
```

Comment:
I would add a step here to tell the reader to create a `script.py` file to hold the code in this step.